java/code/src/com/redhat/rhn/frontend/xmlrpc/saltkey/SaltKeyHandler.java

```diff
@@ -37,25 +40,120 @@ public SaltKeyHandler(SaltKeyUtils saltKeyUtilsIn) {
         this.saltKeyUtils = saltKeyUtilsIn;
     }
 
+    /**
+     * API endpoint to list accepted salt keys
+     * @param loggedInUser the user
+     * @return 1 on success
+     *
+     * @xmlrpc.doc List accepted salt keys
+     * @xmlrpc.param #param("string", "loggedInUser")
+     * @xmlrpc.param #param("string", "minionId")
+     * @xmlrpc.returntype #array_single("string", "Accepted salt key list")
+     */
+    public List<String> acceptedList(User loggedInUser) {
+        try {
+            ensureOrgAdmin(loggedInUser);
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e);
+        }
+        return saltKeyUtils.acceptedSaltKeyList(loggedInUser);
+
+    }
+
+
+    /**
+     * API endpoint to list pending salt keys
+     * @param loggedInUser the user
+     * @return 1 on success
+     *
+     * @xmlrpc.doc List pending salt keys
+     * @xmlrpc.param #param("string", "loggedInUser")
+     * @xmlrpc.returntype #array_single("string", "Pending salt key list")
+     */
+    public List<String> pendingList(User loggedInUser) {
+        try {
+            ensureOrgAdmin(loggedInUser);
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e);
+        }
+        return saltKeyUtils.unacceptedSaltKeyList(loggedInUser);
+    }
+
+    /**
+     * API endpoint to accept minion keys
+     * @param loggedInUser the user
+     * @param minionId the key identifier (minionId)
+     * @return 1 on success
+     *
+     * @xmlrpc.doc Accept a minion key
+     * @xmlrpc.param #param("string", "loggedInUser")
+     * @xmlrpc.param #param("string", "minionId")
+     * @xmlrpc.returntype #return_int_success()
+     */
+    public int accept(User loggedInUser, String minionId) {
+        try {
+            ensureOrgAdmin(loggedInUser);
```

Comment:
As above: This can live outside of the try/catch. Not sure if acceptSaltKey() can throw a PermissionException which needs to be converted. It might be possible when salt master cannot write some files.